### PR TITLE
Adding Unity project configuration to allow GDK work on iOS and OS X

### PR DESCRIPTION
**Pull request by gkassabli**
This diff adds various config changes: downloading iOS worker SDK and respective artifacts, mobile local launch configuration, OS X UnityGameLogic worker configuration and Unity project settings

**Contributions**: We are not currently taking public contributions - see our [contributions](CONTRIBUTING.md) policy. However, we are accepting issues and we do want your [feedback](../README.md#give-us-feedback).

-------

This diff adds various config changes: downloading iOS worker SDK and respective artifacts, mobile local launch configuration, OS X UnityGameLogic worker configuration and Unity project settings
Generate Unity project for iOS and ensure it is runnable. Also ensure that mobile_launch configuration is launchable on OSX and starts UnityGameLogic worker
Will be documented in following diff with iOS build manual
@jessicafalk

### CODE REVIEW

File: ci/download-core-sdk.sh [hunk 1]
```diff
@@ -35,6 +35,7 @@ spatial package retrieve "worker_sdk" "core-dynamic-x86-win32"       "${PINNED_C
 spatial package retrieve "worker_sdk" "core-dynamic-x86_64-win32"    "${PINNED_CORE_SDK_VERSION}" "${CORE_SDK_DIR}/worker_sdk/core-dynamic-x86_64-win32"
 spatial package retrieve "worker_sdk" "core-dynamic-x86_64-linux"    "${PINNED_CORE_SDK_VERSION}" "${CORE_SDK_DIR}/worker_sdk/core-dynamic-x86_64-linux"
 spatial package retrieve "worker_sdk" "core-bundle-x86_64-macos"     "${PINNED_CORE_SDK_VERSION}" "${CORE_SDK_DIR}/worker_sdk/core-bundle-x86_64-macos"
+spatial package retrieve "worker_sdk" "core-dynamic-arm-ios"         "${PINNED_CORE_SDK_VERSION}" "${CORE_SDK_DIR}/worker_sdk/core-dynamic-arm-ios"
```
Comment (jessicafalk) [inline ci/download-core-sdk.sh]: there should also be one for x86 / emulator

File: workers/unity/ProjectSettings/UnityConnectSettings.asset [hunk 2]
```diff
@@ -3,7 +3,7 @@
 --- !u!310 &1
 UnityConnectSettings:
   m_ObjectHideFlags: 0
-  m_Enabled: 0
+  m_Enabled: 1
```
Comment (jessicafalk) [inline workers/unity/ProjectSettings/UnityConnectSettings.asset]: you can revert this file

File: workers/unity/spatialos.UnityGameLogic.worker.json [hunk 3]
```diff
@@ -53,6 +53,29 @@
         "${IMPROBABLE_LOG_FILE}"
       ]
     },
+    "macos": {
+      "artifact_name": "UnityGameLogic@Mac.zip",
```
Comment (jessicafalk) [inline workers/unity/spatialos.UnityGameLogic.worker.json]: nice!

File: ci/download-core-sdk.sh (hunk 1 above)
Comment (gkassabli) [reply ci/download-core-sdk.sh]: Yeah, we'll need it for iOS simulator, but I think getting it working will need a few more things in Unity.
I was planning on creating a separate diff to address that.
Does that make sense?

Comment (jessicafalk) [reply ci/download-core-sdk.sh]: I don't think that there is much more needed on the Unity side, it should just work. But I'm fine with having it in a separate PR

**Comment by jessicafalk** [comment]
should this be merged into `UTY-744-IL2CPP-support`? or into `experimental\mobile-support`?

**Comment by jessicafalk** [comment]
btw: this will need to be redone as soon as the Core C# SDK PR got merged in

**Pull request closed without merging**